### PR TITLE
feat: Allow adding metrics prefix name

### PR DESCRIPTION
**Pull request by klanezurbano**

### CODE REVIEW

**Comment by joao-fontenele** [comment]
@klanezurbano closing due to #9 being merged, thanks for the contribution

**Comment by klanezurbano** [comment]
Updated my pull request to fix merge conflicts

**Comment by klanezurbano** [comment]
@joao-fontenele Updated my pull request to fix merge conflicts

File: README.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ npm i --save express-prometheus-middleware
 | collectDefaultMetrics | Whether or not to collect `prom-client` default metrics. These metrics are usefull for collecting saturation metrics, for example. | `true` |
 | requestDurationBuckets | Buckets for the request duration metrics (in milliseconds) histogram | Uses `prom-client` utility: `Prometheus.exponentialBuckets(0.05, 1.75, 8)` |
 | authenticate | Optional authentication callback, the function should receive as argument, the `req` object and return truthy for sucessfull authentication, or falsy, otherwise. This option supports Promise results. | `null` |
+| prefix | (optional) prefix for the metrics name | |
```
Comment (joao-fontenele) [inline README.md]: just to keep the pattern with the other optional config above

```
Optional prefix for the metrics name | no prefix added |
```

File: src/index.js [hunk 2]
```diff
@@ -58,7 +60,9 @@ module.exports = (userOptions = {}) => {
     // when this file is required, we will start to collect automatically
     // default metrics include common cpu and head usage metrics that can be
     // used to calculate saturation of the service
-    Prometheus.collectDefaultMetrics();
+    Prometheus.collectDefaultMetrics({
+      prefix: options.prefix
```
Comment (joao-fontenele) [inline src/index.js]: missing comma dangle here

File: src/index.js [hunk 3]
```diff
@@ -31,7 +31,9 @@ module.exports = (userOptions = {}) => {
   const app = metricsApp || express();
   app.disable('x-powered-by');
 
-  const requestDuration = requestDurationGenerator(options.requestDurationBuckets);
+  const { metricsPath } = options;
```
Comment (joao-fontenele) [inline src/index.js]: metrics path is already defined above, just removing this line, should fix it

File: src/metrics.js [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
+  });
+}
 
 /**
  * @param {!Array} buckets - array of numbers, representing the buckets for
+ * @param prefix - metrics name prefix
  * request duration
  */
-function requestDurationGenerator(buckets) {
+function requestDurationGenerator (buckets, prefix = '') {
```
Comment (joao-fontenele) [inline src/metrics.js]: also theres an extra space between the function name and the `(`

I've added a linting script to the project `npm run lint`

**Comment by klanezurbano** [comment]
Hi @joao-fontenele you can check my PR again. Thanks.

**Comment by joao-fontenele** [comment]
already merged on #14

**Pull request closed without merging**